docs/source/pipeline.md

```diff
@@ -1,20 +1,36 @@
+Models are composed of three building blocks:
 
-<div style="text-align: center; margin: 20pt"><img src="_images/preproc_data_example.png" alt="Example of preprocessed parquet file" style="width:800px;"/><br><figcaption style="font-style: italic;">Example of preprocessed parquet file</figcaption></div>
+- tokenizer that converts raw text to sparse index encodings
+- transformer architecture
+- head for NLP tasks such as Text Classification, Generation, Sentiment Analysis, Translation, and Summarization. 
 
-NVTabular can easily prepare such data with the [Groupby](https://nvidia-merlin.github.io/NVTabular/main/api/ops/groupby.html) op, which allows grouping by a categorical column (e.g. user id, session id), sorting by another column (e.g. timestamp) and aggregating other columns as sequences (`list`) or by taking the `first` or `last` element of the sequence, as exemplified below.
+Only the transformer architecture building block (b) and their configuration classes are leveraged from HF Transformers. Transformers4Rec provides additional building blocks that are necessary for recommendation, such as input features like normalization and aggregation, as well as heads for recommendation and sequence classification/prediction. These building blocks' `Trainer` class are extended to allow for the evaluation with RecSys metrics.
+
+## Integration with NVTabular
+
+[NVTabular](https://github.com/NVIDIA-Merlin/NVTabular/) is a feature engineering and preprocessing library for tabular data that is designed to easily manipulate datasets at terabyte scale and train deep learning (DL) based recommender systems.
+
+Some popular [techniques](https://nvidia-merlin.github.io/NVTabular/main/api/index.html) have been implemented within NVTabular to deal with categorical and numerical features, such as `Categorify`, `Normalize`, `Bucketize`, `TargetEncoding`, and `DifferenceLag`, and allow for custom transformations (`LambdaOp`) to be defined using cuDF data frame operations.
+
+Typically, the input RecSys datasets contain one example per user interaction. For sequential recommendation, the training example is a sequence of user interactions. For session-based recommendation, the training example is a sequence of session interactions. In practice, each interaction-level feature needs to be converted to a sequence that is grouped by user/session and their sequence length must match since each position of the sequence corresponds to one interaction. To visualize what the preprocessed parquet file should look like, refer to Fig. 2.
```

Comment:
If you are open to suggestion, maybe the last sentence can avoid numbering: "The following figure provides a visualization of the preprocessed tabular data:"

Maybe the figure caption can be changed from "Parquet file" to "Tabular Data".  I don't see the reason to introduce Parquet.